test/parallel/test-timers-unrefed-in-callback.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+// Check that unrefed setInterval timer keeps running even if it is
+// unrefed within callback.
+
+require('../common');
+const assert = require('assert');
+const net = require('net');
+
+let counter1 = 0;
+let counter2 = 0;
+
+// Test1 checks setInterval continues even if it is unrefed within
+// timer callback. counter1 continues to be incremented more than 11
+// until server close completed.
+function Test1() {
+  // server only for maintaining event loop
+  const server = net.createServer().listen(0);
+
+  const timer1 = setInterval(() => {
+    timer1.unref();
+    if (counter1++ === 10)
+      server.close(() => {
+        Test2();
+      });
+  }, 1);
+}
+
+
+// Test2 checks clearInterval also works fine for the unrefed timer
+// within timer callback. counter2 stops at exactly 11 with
+// clearInterval.
+function Test2() {
+  // server only for maintaining event loop
+  const server = net.createServer().listen(0);
+
+  const timer2 = setInterval(() => {
+    timer2.unref();
+    if (counter2++ === 10) {
+      clearInterval(timer2);
+      server.close();
+    }
+  }, 1);
+}
+
+process.on('exit', () => {
+  assert.strictEqual(counter2, 11);
```

Comment:
Here's what I understand of how the test `Test2` works and what its goal is:

* The goal is to make sure that, when calling `clearInterval` on an unrefed() repeating timer created with `setInterval`, the timer doesn't fire anymore.

* To test that, we:

  1. create a repeating timer with `setInterval`
  2. make sure it fires 11 times
  3. clear it
  4. test that it was properly cleared by testing that, when the process exits, the value of `counter2` is 11.

What I'm saying is that, even if the timer _was not_ cleared properly, it would be possible for this test to:
 * not fire the cleared interval's callback (and thus leave the value of `counter2` at `11`)
 * close the server handle
 * let the process exit

if it took less than the time it takes for the interval timer to fire (1ms or more) to close the server and terminate the event loop. In that case, we'd still think the test succeeded, when clearly the implementation would have a bug.

Does that make sense or am I missing something?